Add docs about the FLARE pattern (#102)
LangStream/documentation: pipeline-agents/text-processors/query-vector-db.md

- [Milvus.io](../../configuration-resources/data-storage/milvus.md)
- [JDBC](../../configuration-resources/data-storage/jdbc.md)

### Automatically repeating the query over a list of inputs

> It is possible to perform the same query over a list of inputs. This is useful when you want to perform a query over a list of inputs, for example, a list of questions.

Reviewer: ```suggestion
It is possible to perform the same computation over a list of inputs - for example, a list of questions.
```

You can take the [Flare pattern](../../building-applications/flare-pattern.md) as an example.

In the example below we use the 'loop-over' capability to query the database for each document in the list of documents to retrieve.

```yaml
  - name: "lookup-related-documents"
    type: "query-vector-db"
    configuration:
      datasource: "JdbcDatasource"
      # execute the agent for all the document in documents_to_retrieve
      # you can refer to each document with "record.xxx"
      loop-over: "value.documents_to_retrieve"
      query: |
              SELECT text,embeddings_vector
              FROM documents
              ORDER BY cosine_similarity(embeddings_vector, CAST(? as FLOAT ARRAY)) DESC LIMIT 5
      fields:
        - "record.embeddings"
      # as we are looping over a list of document, the result of the query
      # is the union of all the results
      output-field: "value.retrieved_documents"
```   

When you use "loop-over" it means that the agent executes for each element in a list instead that operating on the whole message.
You use "record.xxx" in order to refer to the current element in the list.

> The snippet above performs and query for each element in in the list "documents_to_retrive" that is expected to be a struct like this:

Reviewer: ```suggestion
The snippet above computes the embeddings for each element in the list "documents_to_retrieve". The list is expected to be a struct like this:
```
